llama_index/indices/vector_store/base.py

```diff
@@ -178,15 +180,23 @@ async def _async_add_nodes_to_index(
         # index struct and document store
         if not self._vector_store.stores_text or self._store_nodes_override:
             for result, new_id in zip(embedding_results, new_ids):
-                index_struct.add_node(result.node, text_id=new_id)
-                self._docstore.add_documents([result.node], allow_update=True)
+                # NOTE: remove embedding from node to avoid duplication
+                node = result.copy()
+                node.embedding = None
+
+                index_struct.add_node(node, text_id=new_id)
+                self._docstore.add_documents([node], allow_update=True)
```

Comment:
Removing embedding from node before saving into index struct and docstore to avoid duplication.